[ROOT-5009] TClass::SetDeclFile: second parameter wrong type

### Check duplicate issues.

- [ ] Checked for duplicates

### Description

From https://its.cern.ch/jira/browse/ROOT-5009

In TClass, the following function:
```cpp
void SetDeclFile(const char* name, int line)
```
stores its second parameter in a Short_t:
```cpp
fDeclFileLine = line;
Short_t	fDeclFileLine	line of class declaration
```

Both GCC and I agree that:
```
/cvmfs/atlas.cern.ch/repo/ATLASLocalRootBase/x86_64/root/5.34.05-x86_64-slc5-gcc4.3/include/TClass.h: In member function &acirc;&euro;&tilde;void TClass::SetDeclFile(const char*, int)&acirc;&euro;&trade;:
/cvmfs/atlas.cern.ch/repo/ATLASLocalRootBase/x86_64/root/5.34.05-x86_64-slc5-gcc4.3/include/TClass.h:342: warning: conversion to &acirc;&euro;&tilde;Short_t&acirc;&euro;&trade; from &acirc;&euro;&tilde;int&acirc;&euro;&trade; may alter its value
``

A Short_t is smaller than an int:
```
Short_t Signed Short integer 2 bytes (short)
```
and thus this function could truncate the received value. Either the type of the second parameter of SetDeclFile should be a Short_t (makes sense since this is what the corresponding GetDeclFile returns), or the internal variable needs to be made an 'int' (and GetDeclFile then needs to be updated to reflect this too).

### Reproducer

As per the description

### ROOT version

Any

### Installation method

Any

### Operating system

Any

### Additional context

_No response_

--- a/core/meta/inc/TClass.h
+++ b/core/meta/inc/TClass.h
@@ -548,7 +548,7 @@ public:
    void               SetContextMenuTitle(const char *title);
    void               SetCurrentStreamerInfo(TVirtualStreamerInfo *info);
    void               SetGlobalIsA(IsAGlobalFunc_t);
-   void               SetDeclFile(const char *name, int line) { fDeclFileName = name; fDeclFileLine = line; }
+   void               SetDeclFile(const char *name, Short_t line) { fDeclFileName = name; fDeclFileLine = line; }
    void               SetDelete(ROOT::DelFunc_t deleteFunc);
    void               SetDeleteArray(ROOT::DelArrFunc_t deleteArrayFunc);
    void               SetDirectoryAutoAdd(ROOT::DirAutoAdd_t dirAutoAddFunc);

